### PR TITLE
Connorwon 214 resource page changes

### PR DESCRIPTION
**Pull request by ConnorWon**
## Description

Frontend changes for new resource backend endpoints

## Updates

[Please enter a brief summary of updates to this PR during code review - if any]

## Checklist

- [x] Ran `npm run lint:fix` to format code
- [x] Requested at least one reviewer
- [x] Connected PR to Trello ticket (steps below)
- [x] Addressed code review comments
- [x] Gained at least one approval for your PR

## Connecting your PR to the corresponding ticket:
- Click on the "GitHub" button found on the right of your Trello ticket
- Choose "Attach PR" from the dropdown list
- Link your GitHub account (if you haven't done so already)
- Navigate to`VCL-content-platform` and choose your PR from the dropdown.

### CODE REVIEW

**Comment by ConnorWon** [comment]
[add resource backend endpoint and functionality](https://trello.com/c/BY2em13c/214-add-resource-backend-endpoint-and-functionality)

File: src/pages/Resources/Page/AddResource/AddResource.tsx [hunk 1]
```diff
@@ -1,25 +1,82 @@
...
+            author: yup
+                .string()
+                .required(TEXT.RESOURCE_PAGE.AUTHOR_EMPTY_ERROR),
+            username: yup
+                .string()
+                .required(),
+            resource_link: yup
+                .string()
+                .matches(
+                    /^((ftp|http|https):\/\/)?(www.)?(?!.*(ftp|http|https|www.))[a-zA-Z0-9_-]+(\.[a-zA-Z]+)+((\/)[\w#]+)*(\/\w+\?[a-zA-Z0-9_]+=\w+(&[a-zA-Z0-9_]+=\w+)*)?$/gm,
```
Comment (ElisaHu1) [inline src/pages/Resources/Page/AddResource/AddResource.tsx:44]: we need to change this to a more general regex format
for example https://github.com/UBC-VCL/VCL-content-platform/pull/177/files is a legit resource link

Comment (ConnorWon) [reply src/pages/Resources/Page/AddResource/AddResource.tsx]: Pushed new commit that should fix this

**Review by ryanyae — APPROVED** [review]
This is some great work, great job Connor

**Pull request merged**